### PR TITLE
HRM configuration and tests.

### PR DESCRIPTION
**Pull request by JoelRoxell**
### Description
- Implements HMR configuration used under development.
- Implemented tests which run via jsdom in the terminal.
- Added pre-push hook.
- Added code coverage.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,10 @@ node_modules
 public
 *.log
 build
+<<<<<<< Updated upstream
```
Comment (LinusLjung) [inline .gitignore]: Unresolved merge conflict

File: pre-push.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+git stash -q --keep-index
```
Comment (LinusLjung) [inline pre-push.sh]: I'd suggest adding the `-u` flag to also stash untracked files

File: pre-push.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+
+# Tests
+npm test
+
+RESULT=$?
+
+git stash pop -q
```
Comment (LinusLjung) [inline pre-push.sh]: You need to check if anything was actually stashed before popping. Otherwise unwanted code could be popped instead

**Review by LinusLjung — CHANGES_REQUESTED** [review]

File: pre-push.sh (hunk 3 above)
Comment (JoelRoxell) [reply pre-push.sh]: Removed the hook implementation. We'll use travic to test builds which will defeat the purpose of pre-tests.

**Review by LinusLjung — APPROVED** [review]

**Pull request merged**